### ISSUE TITLE
Feature Request: Specify order of sourcing R files

### ISSUE THREAD

**Issue by danielinteractive** [pharmaverse/envsetup#70]
### Feature Idea

I did not see a way to specify the sourcing order of R files. In R packages this works via the `@include` roxygen2 tag and subsequent update of collate directive in the DESCRIPTION file. It would be great if `envsetup` would have a similar way to do this.

### Relevant Input

Say you have file `a.R` which for successful sourcing depends on code in `b.R` to be sourced beforehand. This arises e.g. in S4 class definitions or function factory calls. 

### Relevant Output

Currently this will fail. It would be great if there is a way to define that `a.R` depends on `b.R` and therefore `b.R` should be sourced first.

**Comment by nicholas-masel** [comment]
@danielinteractive Thanks for the comment. We should be able to add a tag to each script for this, or we can add a place in the envsetup configuration file that would allow users to overwrite the default source order.

**Comment by danielinteractive** [comment]
Thank you @nicholas-masel , maybe it would even be possible to use a similar pattern as in `roxygen2` by having a look how they get from the `@include` tag usage to the collate list order. The advantage would be to minimize the workload for the user (only need to use such tag when needed)

**Comment by nicholas-masel** [comment]
@danielinteractive  Yep, I completely agree and this looks straightforward.  We'll update `autos()` to call `roxygen:::generate_collate()` to define the source order.

Here is a simple example:

```
#' @include inc2.R inc3.R
inc1 <- function(){
}
```
```

> roxygen2:::generate_collate("tests/testthat/ref/R")
[1] "inc3.R" "inc2.R" "inc1.R"
```

**Comment by nicholas-masel** [comment]
Closed with #71
